Arguments and macro docs update (#7154)
fishtown-analytics/docs.getdbt.com: website/snippets/_validate-macro-args.md

:::tip
From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt will:

> dbt will:

Author: ```suggestion

```

:::tip
From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt will:

> - Raise a warning if `type` fields don’t follow [supported formats](/reference/resource-properties/arguments#supported-types).

Author: ```suggestion

```

:::tip
From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt will:

> - Raise a warning if documented argument names don’t match the macro definition.

Author: ```suggestion

```

:::tip
From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt will:

> - Infer arguments from the macro and includes them in the [manifest.json](/reference/artifacts/manifest-json) file if no arguments are documented.

Author: ```suggestion

```

:::tip
From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt will:

> - Raise a warning if `type` fields don’t follow [supported formats](/reference/resource-properties/arguments#supported-types).

Author: ```suggestion

```

:::tip
From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt will:

> - Raise a warning if documented argument names don’t match the macro definition.

Author: ```suggestion

```

> From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt infers arguments from the macro and includes them in the manifest.json file if no arguments are documented.

Reviewer: sugg adding the infer as a bullet per Grace's sugg. If this is commit is accepted, you'd have to remove the bullets below (lines 4-6) so the bullets aren't duplicated

```suggestion
From dbt Core v1.10, you can opt into validating the arguments you define in macro documentation using the `validate_macro_args` behavior change flag. When enabled, dbt will:

- Infer arguments from the macro and includes them in the [manifest.json](/reference/artifacts/manifest-json) file if no arguments are documented.
- Raise a warning if documented argument names don’t match the macro definition.
- Raise a warning if `type` fields don’t follow [supported formats](/reference/resource-properties/arguments#supported-types).
```

> dbt will:

Reviewer: remove lines 4-6 if the above sugg is accepted

- Raise a warning if `type` fields don't follow [supported formats](/reference/resource-properties/arguments#supported-types).

> Learn more about [macro argument validation](/reference/resource-properties/arguments#supported-types).

Reviewer: ```suggestion
Learn more about [macro argument validation](/reference/global-configs/behavior-changes#macro-argument-validation).
```
